[2025-09-11T03:33:25.611+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: realtime_pipeline.spark_job manual__2025-09-11T03:33:22.297554+00:00 [queued]>
[2025-09-11T03:33:25.623+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: realtime_pipeline.spark_job manual__2025-09-11T03:33:22.297554+00:00 [queued]>
[2025-09-11T03:33:25.624+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-09-11T03:33:25.644+0000] {taskinstance.py:1382} INFO - Executing <Task(BashOperator): spark_job> on 2025-09-11 03:33:22.297554+00:00
[2025-09-11T03:33:25.652+0000] {standard_task_runner.py:57} INFO - Started process 189 to run task
[2025-09-11T03:33:25.658+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'realtime_pipeline', 'spark_job', 'manual__2025-09-11T03:33:22.297554+00:00', '--job-id', '19', '--raw', '--subdir', 'DAGS_FOLDER/realtime_pipeline.py', '--cfg-path', '/tmp/tmp05qnnl8_']
[2025-09-11T03:33:25.664+0000] {standard_task_runner.py:85} INFO - Job 19: Subtask spark_job
[2025-09-11T03:33:25.778+0000] {task_command.py:416} INFO - Running <TaskInstance: realtime_pipeline.spark_job manual__2025-09-11T03:33:22.297554+00:00 [running]> on host 172dbe51b30e
[2025-09-11T03:33:25.910+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='realtime_pipeline' AIRFLOW_CTX_TASK_ID='spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-09-11T03:33:22.297554+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-11T03:33:22.297554+00:00'
[2025-09-11T03:33:25.911+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-09-11T03:33:25.913+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'docker exec spark-master /opt/bitnami/spark/bin/spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.5.0,org.postgresql:postgresql:42.5.1 /opt/bitnami/spark/apps/spark_streaming.py']
[2025-09-11T03:33:25.924+0000] {subprocess.py:86} INFO - Output:
[2025-09-11T03:33:29.818+0000] {subprocess.py:93} INFO - Exception in thread "main" java.lang.IllegalArgumentException: basedir must be absolute: ?/.ivy2/local
[2025-09-11T03:33:29.819+0000] {subprocess.py:93} INFO - 	at org.apache.ivy.util.Checks.checkAbsolute(Checks.java:48)
[2025-09-11T03:33:29.820+0000] {subprocess.py:93} INFO - 	at org.apache.ivy.plugins.repository.file.FileRepository.setBaseDir(FileRepository.java:137)
[2025-09-11T03:33:29.821+0000] {subprocess.py:93} INFO - 	at org.apache.ivy.plugins.repository.file.FileRepository.<init>(FileRepository.java:44)
[2025-09-11T03:33:29.822+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmitUtils$.createRepoResolvers(SparkSubmit.scala:1274)
[2025-09-11T03:33:29.822+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmitUtils$.buildIvySettings(SparkSubmit.scala:1381)
[2025-09-11T03:33:29.823+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.DependencyUtils$.resolveMavenDependencies(DependencyUtils.scala:182)
[2025-09-11T03:33:29.823+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.prepareSubmitEnvironment(SparkSubmit.scala:339)
[2025-09-11T03:33:29.824+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:969)
[2025-09-11T03:33:29.825+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:199)
[2025-09-11T03:33:29.825+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:222)
[2025-09-11T03:33:29.826+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
[2025-09-11T03:33:29.826+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1125)
[2025-09-11T03:33:29.827+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1134)
[2025-09-11T03:33:29.827+0000] {subprocess.py:93} INFO - 	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-09-11T03:33:29.848+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-09-11T03:33:29.870+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 210, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-09-11T03:33:29.877+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=realtime_pipeline, task_id=spark_job, execution_date=20250911T033322, start_date=20250911T033325, end_date=20250911T033329
[2025-09-11T03:33:29.902+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 19 for task spark_job (Bash command failed. The command returned a non-zero exit code 1.; 189)
[2025-09-11T03:33:29.936+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-09-11T03:33:29.967+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
